pkg/operator/resource/resourceapply/apiextensions.go

```diff
@@ -40,3 +40,20 @@ func ApplyCustomResourceDefinitionV1(ctx context.Context, client apiextclientv1.
 
 	return actual, true, err
 }
+
+func DeleteCustomResourceDefinitionV1(ctx context.Context, client apiextclientv1.CustomResourceDefinitionsGetter, recorder events.Recorder, required *apiextensionsv1.CustomResourceDefinition) (*apiextensionsv1.CustomResourceDefinition, bool, error) {
+	existing, err := client.CustomResourceDefinitions().Get(ctx, required.Name, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, false, nil
+		} else {
+			return nil, false, err
+		}
+	}
+	err = client.CustomResourceDefinitions().Delete(ctx, existing.Name, metav1.DeleteOptions{})
```

Comment:
you can save a call by simply calling delete and ignoring NotFound.

Reply:
I believe you don't want me to do a get? I am fine with it.